src/screens/SendSynthetic/SendSyntheticAmount.js

```diff
@@ -17,236 +17,190 @@
-    this.receiverEnsName = getNavigationParam('receiverEnsName');
-
-    this.assetData = getNavigationParam('assetData', {});
-    const fetchedSyntheticAsset = availableSyntheticAssets.find(({ symbol }) => symbol === this.assetData.token);
-    this.availableSyntheticBalance = get(fetchedSyntheticAsset, 'availableBalance', 0);
-    this.syntheticExchangeRate = get(fetchedSyntheticAsset, 'exchangeRate', 0);
-
-    const intentError = !this.availableSyntheticBalance
-      ? 'Asset has no available liquidity'
-      : null;
 
     this.state = {
-      intentError,
+      intentError: '',
       submitPressed: false,
       value: null,
       inputHasError: false,
+      selectedContact: null,
+      assetData: null,
+      receiver: '',
+      receiverEnsName: '',
     };
 
-    this.handleFormChange = debounce(this.handleFormChange, 500);
+    this.handleAssetValueSelect = debounce(this.handleAssetValueSelect, 500);
   }
 
   componentDidMount() {
-    this.props.initSyntheticsService();
-    this.props.fetchSingleAssetRates(this.assetData.token);
+    const { initSyntheticsService, fetchAvailableSyntheticAssets } = this.props;
+    initSyntheticsService();
+    fetchAvailableSyntheticAssets();
+    this.setPreselectedValues();
   }
 
+  setPreselectedValues = () => {
+    const { navigation, contactsByWallet } = this.props;
+    const contact = navigation.getParam('contact');
+    if (contact) {
+      const formattedContact = contactsByWallet.find(({ name }) => name === contact.username);
+      if (formattedContact) this.setReceiver(formattedContact);
+    }
+  };
+
+  handleReceiverSelect = (value: Option, onSuccess?: () => void) => {
+    const { navigation } = this.props;
+    const { name, hasSmartWallet } = value;
+
+    if (hasSmartWallet) {
+      this.setReceiver(value, onSuccess);
+    } else {
+      Alert.alert(
+        'This user is not on Pillar Network',
+        'You both should be connected to Pillar Network in order to be able to send instant transactions for free',
+        [
+          { text: 'Open Chat', onPress: () => navigation.navigate(CHAT, { username: name }) },
+          { text: 'Switch to Ethereum Mainnet', onPress: () => navigation.navigate(ACCOUNTS) },
+          { text: 'Cancel', style: 'cancel' },
+        ],
+        { cancelable: true },
+      );
+    }
+  };
 
-  checkFormInputErrors = () => {
-    if (!this.syntheticsForm) return;
-    const { inputHasError } = this.state;
-    if (!isEmpty(get(this.syntheticsForm.validate(), 'errors'))) {
-      this.setState({ inputHasError: true });
-    } else if (inputHasError) {
-      this.setState({ inputHasError: false });
+  setReceiver = async (value: Option, onSuccess?: () => void) => {
+    const { receiverEnsName, receiver } = await getContactsEnsName(value?.ethAddress);
+    let stateToUpdate = {};
+    if (!receiver) {
+      Toast.show({
+        title: 'Invalid ENS Name',
+        message: 'Could not get address',
```

Comment:
Let's be a bit nicer here... this is our grandma after all! 👵 

```suggestion
        title: 'Sorry, we could not find that ENS name!',
        message: 'Could you please check and try again?',
```

Reply:
Thank you - it's the little things in life 😄